include/seqan3/alphabet/adaptation/char.hpp

```diff
@@ -27,117 +27,53 @@
-                                               char16_t,
-                                               char32_t,
-                                               wchar_t>;
-// TODO Windows adaption requires different handling of wchar_t, in Windows it is 16 bits and holds UTF-16 code units
-//!\brief The corresponding list of rank types
-using char_adaptations_rank_types = meta::list<std::uint_least8_t,
-                                               std::uint_least16_t,
-                                               std::uint_least32_t,
-                                               std::uint_least32_t>;
-
-//!\brief Metafunction overload for types that are in seqan3::detail::char_adaptations.
-template <typename type_in_list>
-//!\cond
-    requires meta::in<char_adaptations, std::remove_reference_t<type_in_list>>::value
-//!\endcond
-struct is_char_adaptation<type_in_list> :
-    std::true_type
-{};
+//!\brief Whether a type is `char`, `char16_t`, `char32_t` or `wchar_t`.
+template <typename type, typename type_no_ref = std::remove_reference_t<type>>
+constexpr bool is_char_adaptation_v = std::Same<type_no_ref, char>     ||
+                                      std::Same<type_no_ref, char16_t> ||
+                                      std::Same<type_no_ref, char32_t> ||
+                                      std::Same<type_no_ref, wchar_t>;
 
 //!\}
 } // namespace seqan3::detail
 
-namespace seqan3
+namespace seqan3::adaptation
 {
 
-//!\addtogroup adaptation
-//!\{
-
-// ------------------------------------------------------------------
-// type metafunctions
-// ------------------------------------------------------------------
-
-/*!\brief Specialisation of seqan3::underlying_char for char types.
- * \tparam char_type One of `char`, `char16_t`, `char32_t` or `wchar_t`.
- * \relates seqan3::char_adaptation_concept
- * \sa seqan3::alphabet_char_t
+/*!\name Free function wrappers for the char alphabet adaptation
+ * \brief For `char`, `char16_t` and `char32_t` do conversion to/from uint types.
+ * \ingroup adaptation
+ * \{
  */
-template <typename char_type>
-//!\cond
-    requires detail::is_char_adaptation_v<char_type>
-//!\endcond
-struct underlying_char<char_type>
-{
-    //!\brief The same type as char_type.
-    using type = char_type;
-};
 
-/*!\brief Specialisation of seqan3::underlying_rank for char types.
+/*!\brief Return the number of values the char type can take.
  * \tparam char_type One of `char`, `char16_t`, `char32_t` or `wchar_t`.
- * \relates seqan3::char_adaptation_concept
- * \sa seqan3::alphabet_rank_t
+ * \param chr The parameters actual value is ignored.
+ * \returns The respective sizes (e.g. 256 for `char`).
  */
 template <typename char_type>
-//!\cond
-    requires detail::is_char_adaptation_v<char_type>
-//!\endcond
-struct underlying_rank<char_type>
+constexpr auto alphabet_size(char_type const & SEQAN3_DOXYGEN_ONLY(chr)) noexcept
+    requires ::seqan3::detail::is_char_adaptation_v<char_type>
```

Comment:
Sometimes doxygen displays this correctly if you put it after the function declarator.  
But it seems to be confusing what rules should be applied. Like, when to surround it with cond or not. I would generally say, we need to wait for concepts support in doxygen (really, don't know, if this will ever happen) and should rather put for all versions the stuff into a cond block

Reply:
AFAICT parens and braces in the term break doxygen's handling of the trailing requires statement, but I can still add the \cond.

Reply:
I'll move them to the top then as well